graph/run-dynamic.go

```diff
@@ -263,59 +263,69 @@ func ConvergeDynamicThread[EP EPP[E], V VPI[V], E EPI[E], M MVI[M], N any, A Alg
 				// There is a request for synchronously syncing topology (blocking), acknowledge here as we have no more topology events to process.
 				bspSync = false
 				gt.Response <- ACK
-			} else if remitCount == 0 {
-				// Want topology events, but none found -- we should relax.
-				topFail++
-				gt.Status = BACKOFF_TOP
-				utils.BackOff(topFail)
-				if timeStates && tidx == 0 {
-					curr = time.Now()
-					gt.LoopTimes[5] += curr.Sub(prev)
-					prev = curr
-				}
-			} else {
-				topFail = 0
+			} else if remitCount == 0 && !epoch {
+				topFailed = true
 			}
 
 			if strucClosed {
 				doneEvents <- struct{}{}
-			} else if TOPOLOGY_FIRST || blockAlgIfTop {
-				continue
 			}
 		}
 
-		// Process algorithm messages. Check for algorithm termination if needed.
-		gt.Status = APPLY_MSG
-		checkTerm := (strucClosed && remitClosed) || (epoch && remitCount == 0 && count == 0)
-		completed, count = ProcessMessages[V, E, M, N](alg, g, gt, checkTerm)
-		if completed && epoch {
-			//log.Debug().Msg("T[" + utils.F("%02d", tidx) + "] completed epoch")
-			gt.Status = DONE
-			gt.Response <- ACK
-
-			resp := <-gt.Command // BLOCK and wait for resume
-			if resp != RESUME {
-				log.Panic().Msg("Expected to resume after blocked")
+		algFailed := false
```

Comment:
```suggestion
		algFailed := true
```

Reply:
It seems like this could trigger backup when there is no need to process algorithmic messages.

Reply:
I suggest we distinguish between success, failure, and no need to process algorithmic messages

Reply:
Yep, that is actually the goal. Check the performance with static loading of a larger graph and you'll see quite a difference if there's no backoff.